### PR TITLE
Fix the top command on Windows

### PR DESCRIPTION
**Pull request by dardelean**
The top command did not work for Windows before, the `containerTop()` function was not implemented.

Signed-off-by: Dan Ardelean [dardelean@cloudbasesolutions.com](mailto:dardelean@cloudbasesolutions.com)

### CODE REVIEW

File: pkg/cmd/container/top.go [hunk 1]
```diff
@@ -203,83 +198,3 @@ func parsePSOutput(output []byte, procs []uint32) (*ContainerTopOKBody, error) {
 	}
 	return procList, nil
 }
-
```
Comment (aznashwan) [inline pkg/cmd/container/top.go:206]: You'll need to add a definition for `containerTop` on FreeBSD too please.

If it's identical to the Linux one, you can simply rename the `pkg/cmd/container/top_linux.go` file to `pkg/cmd/container/top_unix.go` and simply add a `!windows` build tag.

**Comment by AkihiroSuda** [comment]
CI failing

```
These files were modified:
  
   M go.mod
  
  Content of the diff:
  diff --git a/go.mod b/go.mod
  index 89981b1..ee26559 100644
  --- a/go.mod
  +++ b/go.mod
  @@ -61,12 +61,12 @@ require (
   require (
   	github.com/AdaLogics/go-fuzz-headers v0.0.0-20230106234847-43070de90fa1 // indirect
   	github.com/AdamKorcz/go-118-fuzz-build v0.0.0-20221215162035-5330a85ea652 // indirect
  -	github.com/Microsoft/hcsshim v0.10.0-rc.5 // indirect
  +	github.com/Microsoft/hcsshim v0.10.0-rc.5
   	github.com/cilium/ebpf v0.9.1 // indirect
   	github.com/containerd/cgroups/v3 v3.0.0 // indirect
   	github.com/containerd/fifo v1.0.0 // indirect
   	github.com/containerd/ttrpc v1.1.1-0.20230127163717-32fab2374638 // indirect
  -	github.com/containerd/typeurl v1.0.3-0.20220422153119-7f6e6d160d67 // indirect
  +	github.com/containerd/typeurl v1.0.3-0.20220422153119-7f6e6d160d67
   	github.com/containers/ocicrypt v1.1.6 // indirect
   	github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
   	github.com/distribution/distribution/v3 v3.0.0-2023021[415](https://github.com/containerd/nerdctl/actions/runs/4232290817/jobs/7351844087#step:4:424)0026-36d8c594d7aa // indirect
```

File: pkg/cmd/container/top_windows.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/Microsoft/hcsshim/cmd/containerd-shim-runhcs-v1/options"
+	"github.com/containerd/containerd"
+	"github.com/containerd/typeurl"
```
Comment (AkihiroSuda) [inline pkg/cmd/container/top_windows.go:30]: Can we use v2?
https://github.com/containerd/nerdctl/blob/6d085c1c414bc76b5298b0153dd6379f4f22b671/go.mod#L20

Comment (dardelean) [reply pkg/cmd/container/top_windows.go]: Done.

File: cmd/nerdctl/container_top_unix_test.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//go:build linux || darwin || freebsd || netbsd || openbsd
```
Comment (dcantah) [inline cmd/nerdctl/container_top_unix_test.go:1]: Could we do !windows?

Comment (AkihiroSuda) [reply cmd/nerdctl/container_top_unix_test.go:1]: Probably you can just drop this line, as `unix` has been a built-in tag since Go 1.19

Comment (dcantah) [reply cmd/nerdctl/container_top_unix_test.go:1]: Oh even better, didn't know that

**Review by dcantah — APPROVED** [review]
lgtm, lets use the unix build tag convenience though

File: cmd/nerdctl/container_top_windows_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		t.Fatalf("unable to access Windows service control manager: %s", err)
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(hypervServiceName)
+	// hyperv service is not present, hyperv is not enabled
+	if err != nil {
+		t.Skip("HyperV is not enabled, skipping test")
+	}
+	defer s.Close()
```
Comment (dcantah) [inline cmd/nerdctl/container_top_windows_test.go:51]: I'm adding a little helper we can use for future hyperv tests here https://github.com/containerd/nerdctl/pull/2062

File: cmd/nerdctl/container_top_unix_test.go (hunk 3 above)
Comment (dardelean) [reply cmd/nerdctl/container_top_unix_test.go:1]: If the directive is not used, I get errors like this when running tests "cmd\nerdctl\container_top_test.go:38:62: undefined: testutil.AlpineImage".

**Comment by dardelean** [comment]
Ping.

**Pull request merged**